File: src/main/java/test_lea_0003/a02_divisorSubstrings/solution.java
public class solution {
    public int divisorSubstrings(int n, int k) {
        return sumOfDivisors(n, k);
    }
    
    private int sumOfDivisors(int inputData, int groupLength) {
        int result = 0;
        if (0 == inputData) {
            return result;
        }
        List<Integer> divisors = getDivisorsToTry(inputData, groupLength);
        for (int i = 0; i < divisors.size(); i++) {
            
        }
        return result;
    }
    
    private List<Integer> getDivisorsToTry(int inputData, int groupLength) {
        final String inputString = String.valueOf(inputData);
        List<Integer> resultList = new ArrayList<>();
        for (int i = 0; i < inputString.length() - 1; i++) {
            System.out.println("input data " + inputData);
            System.out.println("iteration = " + i);
            System.out.println("gorup length = " + groupLength);
            System.out.println("substring = " + inputString.substring(i, groupLength));
            // resultList.add(Integer.valueOf(inputString.substring(i, groupLength)));
        }
        return resultList;
    }
}

File: src/main/java/test_lea_0003/a02_divisorSubstrings/solution.py
def sum_of_divisors(input_data: int, group_length: int):
    if input_data == 0:
        return 0
    input_string = str(input_data)
    result_list = []
    for index in range(0, len(input_string) - group_length + 1):
        potential_result: int = int(input_string[index:index + group_length])
        if input_data % potential_result == 0:
            result_list.append(potential_result)
    return len(set(result_list))
